tests/test_zarr_ipfs.py

```
@@ -104,3 +104,36 @@ def test_upload_then_read(random_zarr_dataset: tuple[str, xr.Dataset]):
+    loaded_ds = xr.open_zarr(store=hamt)
+    xr.testing.assert_identical(loaded_ds, expected_ds)
+
+    # Now trying to load without a decryptor, xarray should be able to read the metadata and still perform operations on the unencrypted variable
+    print("Attempting to read and print metadata of partially encrypted zarr")
+    ds = xr.open_zarr(
+        store=HAMT(store=IPFSStore(), root_node_id=hamt.root_node_id, read_only=True)
+    )
+    print(ds)
+    assert ds.temp.sum() == expected_ds.temp.sum()
+    # We should be unable to read precipitation values which are still encrypted
+    with pytest.raises(Exception):
```

Ruff (0.8.2)

`pytest.raises(Exception)` should be considered evil

(B017)